Conan.VisualStudio/ConanOptionsPage.cs

```diff
@@ -12,14 +15,46 @@ public class ConanOptionsPage : DialogPage
         private bool? _conanInstallAutomatically;
         private ConanBuildType? _conanBuild;
         private bool? _conanUpdate;
+        private static bool ValidateConanExecutable(string exe)
+        {
+            if (exe == null || exe == "")
+                return true;
+            try
+            {
+                var startInfo = new ProcessStartInfo
+                {
+                    FileName = exe,
+                    Arguments = "--version",
+                    UseShellExecute = false,
+                    RedirectStandardInput = true,
+                    RedirectStandardOutput = true,
+                    CreateNoWindow = true
+                };
+                var process = Process.Start(startInfo);
+
+                process.WaitForExit();
+
+                if (0 != process.ExitCode)
+                    MessageBox.Show($"invalid conan executable {exe}: conan --version failed with error {process.ExitCode}",
+                        "invalid conan executable", MessageBoxButtons.OK, MessageBoxIcon.Error);
+
+                return 0 == process.ExitCode;
+            }
+            catch (Win32Exception e)
+            {
+                MessageBox.Show($"invalid conan executable {exe}: {e.Message}", "invalid conan executable",
+                    MessageBoxButtons.OK, MessageBoxIcon.Error);
+                return false;
+            }
+        }
 
         [Category("Conan")]
         [DisplayName("Conan executable")]
         [Description(@"Path to the Conan executable file, like C:\Python27\Scripts\conan.exe")]
         public string ConanExecutablePath
         {
             get => _conanExecutablePath ?? (_conanExecutablePath = ConanPathHelper.DetermineConanPathFromEnvironment());
-            set => _conanExecutablePath = value;
+            set => _conanExecutablePath = ValidateConanExecutable(value) ? value : _conanExecutablePath;
```

Comment:
I think we should validate the path returned by `ConanPathHelper.DetermineConanPathFromEnvironment()` too. I'm not sure if this is the place to do this, or it is better to validate it inside the function before returning (return `null` if invalid)

Reply:
I think it should be done inside function then

Reply:
Sure, please add that check

Reply:
added